sentinel/src/schedule/reminders.js

```diff
@@ -108,7 +108,7 @@ const getPlaceIds = (keys, startDocId) => {
   // using `request` library because PouchDB doesn't support `start_key_doc_id` in view queries
   // using `start_key_doc_id` because using `skip` is *very* slow
   return request
-    .get(`${db.couchUrl}/_design/medic-client/_view/contacts_by_type`, { qs: query, json: true })
+    .get({ url: `${db.couchUrl}/_design/medic-client/_view/contacts_by_type`, qs: query, json: true })
```

Comment:
How confident are you that there's nowhere else in the repo that calls rpn with the URL string as the first param? I'm worried there's somewhere else that maybe doesn't have test coverage which will now fail...

Reply:
I think I'd feel more comfortable if we changed the implementation to support passing the URL as a string so it's more compatible with the RPN way.

Reply:
```suggestion
  How confident are you that there's nowhere else in the repo that calls rpn with the URL string as the first param? I'm 
  worried there's somewhere else that maybe doesn't have test coverage which will now fail...
```
I've been through it with a fine toothed comb, but can do so again and report.

> I think I'd feel more comfortable if we changed the implementation to support passing the URL as a string so it's more compatible with the RPN way

If you're resolute on this I can change it, but:
- it makes the code more complex
- passing an object is a better API as everything is named and can't be passed in the wrong order. It also makes the code robust to any future API changes that require call site changes (as this one does). I'd argue that it would be a good pattern to enforce over the first param pattern. Libs like rpn will have prioritised ease of use over correctness and robustness, but we don't have to.

Let me know if I've changed your mind or not.

Reply:
For API design it's better to have required parameters as explicit and optional ones in an object. In this case only the URL is required, so I would argue that the better API design is to remove the URL from the object completely.

Ultimately (once it's supported in Node) we'll switch to using the native `fetch` spec ( https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch ) which does it this way (eg: `fetch(url, options)`) so I'd prefer to align with that standard rather than enforcing our own.

For backwards compatibility I can live with supporting both the standard way, and having the url in the object, but long term we should standardise on the HTML spec.

Reply:
Disagree on the pattern, as I've seen too many bugs from people changing param ordering both at call site and arguments. I agree with using the low-level API the way you say, but personally wouldn't support it up the stack as we don't have the same kind of API guarantees with code we own as we do with the spec (depending on who is reviewing).

But it's your rodeo -- will make the change. It's a subtle disagreement anyway and hopefully non-consequential.